web_widget_image_webcam/static/src/js/webcam_widget.js

```diff
@@ -0,0 +1,140 @@
+/* global Webcam */
+/*
+    Copyright 2016 Siddharth Bhalgami <siddharth.bhalgami@gmail.com>
+    License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+*/
+odoo.define("web_widget_image_webcam.webcam_widget", function (require) {
+    "use strict";
+
+    var core = require("web.core");
+    var rpc = require("web.rpc");
+    var Dialog = require("web.Dialog");
+    const {patch} = require("@web/core/utils/patch");
+    const {ImageField} = require("@web/views/fields/image/image_field");
+
+    var _t = core._t;
+    var QWeb = core.qweb;
+
+    patch(ImageField.prototype, "web_widget_image_webcam", {
```

Comment:
Naive question : why do you replace "include" by "patch"? 
Is this feature overloadable ? 

thanks !

Reply:
Hi @legalsylvain, I just wanted to use this 'patch' utility provided by Odoo described here https://www.odoo.com/documentation/16.0/developer/reference/frontend/patching_code.html#

about the rest, I suppose that functions added to ImageFIeld can be overloaded following the same procedure.

Reply:
@ivantodorovich any point of view regarding the topic patch vs include/extend ?

Reply:
The change seems unnecessary whilst the object to patch extends from `OdooClass` (`web.Class`)
But this is super old code and odoo is moving out from this approach to a pure js class AFAIK, so we'll probably see the use of `patch` more and more